### DISCUSSION TITLE
Feature tag for "running game"

### DISCUSSION THREAD

**Discussion by wareya** [godotengine/godot-proposals#4611] [Engine Core]
Using the "Project Settings Override" file, I can get certain settings, configured by the user ingame, to load on engine boot, before the first scene loads. This is good and important, because some settings, like fullscreen, HiDPI support, etc. need to be loaded when the game boots, and not from gdscript code after it boots.

But this file loads when the editor loads, too. Which is bad, because then it overrides the project's project settings.

Okay, in that case I guess I need to use feature tags to make it so that the settings are only loaded by a running copy of the game, and not the editor.

Except, uh. There's no feature tag for that. There is no feature tag that applies when running the game, even when running it through the editor, but not when loading the project in the editor. Nor is there a set of feature tags that could be used in parallel to the same effect.

There should be one.

**Comment by wareya** [comment]
To clarify:

The "debug" and "editor" feature tags are used both when running the editor itself and when running the game from the editor.

And the "standalone" tag is not used when running the game from the editor.

In other words, the following code:

```gdscript
    print(OS.has_feature("debug"))
    print(OS.has_feature("editor"))
    print(OS.has_feature("release"))
    print(OS.has_feature("standalone"))
```

behaves exactly the same in a tool script in the editor as it does in a game running from the editor.

And what I'm asking for here is a feature tag (or pair) that reflects `Engine.editor_hint`, and is handled correctly at engine boot time, even when running the game from the editor.

**Comment by Calinou** [comment]
Feature tags are currently mostly used for things that remain *constant* within a specific compiled binary (with the exception of feature tags for texture compression support).

It's likely possible to add `editor_hint` and `not_editor_hint` feature tags, but these wouldn't be constant and defined at compile-time as a single binary would be able to satisfy both cases depending on how it's run.

That said, I think the Project Settings Override file should be ignored in if `Engine.is_editor_hint()` evaluates to `true`. It would still apply when *running* the project from the editor, but it wouldn't apply to the editor itself.

**Reply by wareya** [reply]
> That said, I think the Project Settings Override file should be ignored in if `Engine.is_editor_hint()` evaluates to true.

For some reason, I was under the impression that there were genuine use-cases for loading it even in the editor, but now that you mention it, there might not be. Everything that comes to mind could be done just as well with `override.cfg` or normal feature tag setting overrides set directly in `project.godot`.

Simply not loading the Project Settings Override file (as specified in the project settings) when running the editor itself would fix my problem here, if it's a possible option.

**Reply by wareya** [reply]
(Having said that, the documentation should probably outline the gotcha that the `editor` tag is true even when running the game from the editor, i.e. that it doesn't correspond to `Engine.editor_hint`.)
